osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs

```diff
@@ -198,11 +207,29 @@ private void updateDisplay()
             updateDifficultyButtons();
         }
 
-        private void showBeatmap(IBeatmapInfo? beatmapInfo)
+        private void showBeatmap(APIBeatmap? beatmapInfo)
         {
+            guestMapperContainer.Clear();
+
+            if (beatmapInfo?.AuthorID != BeatmapSet?.AuthorID)
+            {
+                APIUser? user = BeatmapSet?.RelatedUsers?.SingleOrDefault(u => u.OnlineID == beatmapInfo?.AuthorID);
+                if (user != null)
+                    getGuestMapper(user);
+            }
+
             version.Text = beatmapInfo?.DifficultyName ?? string.Empty;
         }
 
+        private void getGuestMapper(APIUser user)
+        {
+            guestMapperContainer.With(d =>
+            {
+                d.AddText("mapped by ");
+                d.AddUserLink(user);
+            });
+        }
```

Comment:
For future reference, this is almost criminal to me. A `getGuestMapper()` method name suggests that it will return something and be a pure method. Instead, it is a `void` method, so it returns nothing, and it *also* has baffling side effects which are - even more bafflingly - applied via `.With()`, which is supposed to be syntactic sugar first and foremost.

I'll fix it myself to speed proceedings up, but just for future reference I guess.